data2rdf/models/mapping.py

```diff
@@ -0,0 +1,234 @@
+"""Mapping models for data2rdf"""
+import json
+import warnings
+from abc import abstractmethod
+from typing import Any, Dict, Optional, Union
+
+from pydantic import AnyUrl, BaseModel, Field, ValidationInfo, field_validator
+from rdflib import Graph
+
+from data2rdf.config import Config
+from data2rdf.qudt.utils import _get_query_match
+from data2rdf.utils import is_float, is_integer, make_prefix
+
+
+class BasicConceptMapping(BaseModel):
+    """Basic mapping for any entity"""
+
+    key: Optional[str] = Field(
+        None, description="Key of the concept in the file"
+    )
+    iri: AnyUrl = Field(
+        ..., description="Ontological class related to this concept"
+    )
+    config: Config = Field(
+        default_factory=Config, description="Configuration object"
+    )
+
+    @property
+    def suffix(cls) -> str:
+        """Return suffix for individal"""
+        return str(cls.iri).split(cls.config.separator)[-1]
+
+    @field_validator("config")
+    @classmethod
+    def validate_config(cls, value: Union[Dict[str, Any], Config]) -> Config:
+        """Validate configuration"""
+        if isinstance(value, dict):
+            value = Config(**value)
+        return value
+
+
+class ClassConceptMapping(BasicConceptMapping):
+    """Mapping for a concept coming from the mapping file"""
+
+    unit: Optional[Union[str, AnyUrl]] = Field(
+        None, description="Symbol or QUDT IRI for the mapping"
+    )
+    annotation: Optional[Union[str, AnyUrl]] = Field(
+        None, description="Base IRI with which the value shall be concatenated"
```

Comment:
Why use concatenation and not simply provide the full IRI?

Reply:
This is because we want to concatenate with the value of the respective concept.

e.g. 
value ="DP800"
annotation = "https://w3id.org/steel/ProcessOntology/"
-> iri = https://w3id.org/steel/ProcessOntology/DP800